bugsnag-spring/build.gradle

```diff
@@ -0,0 +1,23 @@
+apply plugin: 'java-library'
+apply plugin: 'checkstyle'
+
+sourceCompatibility = 1.6
+targetCompatibility = 1.6
+
+repositories {
+    jcenter()
+}
+
+dependencies {
+}
```

Comment:
I think there are two things we might want to support here as a dependency:

1. the local version of the 'bugsnag-java' module
2. a mavenCentral artefact of bugsnag-java

I suppose the key question is which of those to use when we're making a release of bugsnag-spring. I think the advantages of using a maven artefact would be there's no potential for bugsnag-java to get out of date with bugsnag-spring fixes. Although that would mean we always have to release bugsnag-java first, and wait for it to propagate.

What were your thoughts on this?

Reply:
My personal preference would be to have the local version of the 'bugsnag-java' module as a dependency so that we don't have to wait. It would also make development of the bugsnag-spring module more straightforward when changes also need to be made to bugsnag-java.
You are right about bugsnag-java potentially getting out of date. I'm not sure of the best way around this.

Reply:
Another thing - if a project had a dependency on both bugsnag-java and bugsnag-spring, then this approach could potentially lead to some nasty issues with the classloader loading subtly different class definitions. I think this could be possible if someone misread our upgrade instructions, or if a Spring project depends on some 3rd party library that depends on bugsnag-java. I'm not sure that this scenario would be common enough to swing the argument one way or another.

We may be able to solve bugsnag-java getting out of date via tooling (e.g. checking the latest commit hash on master via a gradle plugin, before releasing bugsnag-spring).

Reply:
How long does it usually take maven releases to propagate?

Reply:
1-2 hours is typical, it can take longer however

Reply:
What about something like this as a compromise:
```
dependencies {
    if (project.hasProperty('releasing')) {
        compile 'com.bugsnag:bugsnag:3.+'
    } else {
        compile project(':bugsnag-java')
    }
}
```

Reply:
To keep consistent with the NDK rewrite approach, we are going to go with `compile project(':bugsnag-java')`.